docs/getting_started.rst

```diff
@@ -31,105 +39,467 @@ example to construct such a store for the local filesystem.
+
+    In this example, we create a dataset with two partitions (represented by
+    the dictionary objects inside the list).
+    For each partition, there exist two tables: ``core-table`` and ``aux-table``.
+    The schemas of the tables are identical across partitions.
+
+    .. ipython:: python
+
+       dfs = [
+            {
+                "data": {
+                    "core-table": pd.DataFrame({"col1": ["x"]}),
+                    "aux-table": pd.DataFrame({"f": [1.1]}),
+                },
+            },
+            {
+                "data": {
+                    "core-table": pd.DataFrame({"col1": ["y"]}),
+                    "aux-table": pd.DataFrame({"f": [3.2]}),
+                },
+            },
+       ]
+
+       store_dataframes_as_dataset(store_factory, dataset_uuid="two-tables", dfs=dfs)
+
+
+Reading dataset from storage
+=============================
+
+After we have written the data, we may want to read it back in again. For this we can
+use :func:`kartothek.io.eager.read_table`. This method returns the complete
+table of the dataset as a pandas DataFrame (since there is only a single table in this
+example, it returns the entire dataset).
+
+.. ipython:: python
+
+    from kartothek.io.eager import read_table
+
+    read_table("a_unique_dataset_identifier", store_factory, table="table")
+    
+
+We can also read a dataframe iteratively, using 
+:func:`kartothek.io.iter.read_dataset_as_dataframes__iterator`. This will return a generator
+of dictionaries (one dictionary for each `partition`), where the keys of each dictionary
+represent the `tables` of the dataset. For example,
+
+.. ipython:: python
+
+    from kartothek.io.iter import read_dataset_as_dataframes__iterator
+
+    for partition_index, df_dict in enumerate(
+            read_dataset_as_dataframes__iterator(dataset_uuid="two-tables", store=store_factory)
+        ):
+            print(f"Partition #{partition_index}")
+            for table_name, table_df in df_dict.items():
+                print(f"Table: {table_name}. Data: \n{table_df}")
+
+
+.. admonition:: Filtering the dataset using predicates
+
+    It is possible to perform reading queries similar to an SQL ``WHERE`` statement using
+    the ``predicates`` argument. When this argument is defined, ``kartothek`` 
+    uses the Apache Parquet metadata to load only chunks of the dataset which may contain values
+    that fullfill the query. Such a query will be significantly faster if the dataset is
+    partitioned or has an index built on the column queried.
+
+    .. ipython:: python
+
+        # Read only values table `aux-table` where `f` < 2.5
+        read_table(
+            "two-tables", store_factory, table="aux-table", predicates=[[("f", "<", 2.5)]]
+        )
+
+
+
+Updating an existing dataset
+============================
+
+It's possible to update datasets by adding new physical partitions to them, ``kartothek`` provides
```

Comment:
It's not only possible to update a dataset by appending but it is also possible to remove/replace partitions using the `delete_scope` argument. It would be nice to have an example for how this works

Reply:
Good point. Will add an example for this.